nifi-api/src/main/java/org/apache/nifi/registry/VariableRegistryFactory.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.registry;
+
+import java.io.File;
+import java.nio.file.Path;
+import java.util.Map;
+import java.util.Properties;
+
+public class VariableRegistryFactory {
+
+    public static VariableRegistry getInstance(final Properties...properties){
+        return new PropertiesVariableRegistry(properties);
+    }
+
+    public static VariableRegistry getInstance(final Path... paths){
```

Comment:
Does it make sense to perhaps use a different name than 'getInstance' here? I suggest this only because the File Variable Registry is abstract, so there could potentially be multiple implementations, so PropertiesVariableRegistry may not always be the correct implementation to use. Perhaps call it `getPropertiesInstance` or something like that? Same for `File...` argument below.